onnxscript/evaluator.py

```diff
@@ -15,7 +18,62 @@
     InvalidGraph,
 )
 
-from onnxscript import irbuilder, tensor, utils
+from onnxscript import autocast, irbuilder, onnx_opset, tensor, utils, values
+
+
+class Evaluator(ABC):
+    """Base class for evaluation of ONNX ops.
+
+    The execution of onnxscript functions in eager-mode is dispatched to an Evaluator
+    instance (or, more precisely, to the eval method of the Evaluator instance).
+    The evaluator is expected to transform the input/output/attribute representation
+    supported by onnxscript to those expected by a particular backend.
+    """
+
+    def eval(self, schema, inputs, attributes):
```

Comment:
Add types maybe?

Reply:
Good point. Agree this is very important (formally pinning down the representation-types used by the different evaluators is essential, I think). But some of these is in flux, I'd prefer to do it a bit later.